tests/test_config.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from pathlib import Path
+
+import pytest
+from pytest import LogCaptureFixture, MonkeyPatch
+
+from lakefs_spec.config import LakectlConfig
+
+
+def test_config_read_nonexistent_file():
+    """
+    Tests that a FileNotFoundError is raised when a lakectl config is read from
+    a nonexistent file.
+    """
+    with pytest.raises(FileNotFoundError):
+        LakectlConfig.read("aaaaaaaaaaaaaaaa.yaml")
+
+
+def test_lakectl_config_parsing_without_yaml(
+    monkeypatch: MonkeyPatch, caplog: LogCaptureFixture, temporary_lakectl_config: str
+) -> None:
+    # unset YAML module from sys.modules.
+    monkeypatch.setitem(sys.modules, "yaml", None)
+
+    LakectlConfig.read(Path("~/.lakectl.yaml").expanduser())
+
+    # look for the log record, assert that it has the `pyyaml` install instruction.
+    assert len(caplog.records) > 0
```

Comment:
Looking at this, I wonder, if the [`warnings` module from the standard library](https://docs.python.org/3/library/warnings.html) would be useful for situations like this. I'll play around with it a bit (but not for this PR, which lgtm)

Reply:
It probably is, I remember correcting a PR like this from a `logger.warning` to a `warnings.warn` to keep a clean stdout / stderr division.

Reply:
There's even [`logging.captureWarnings`](https://docs.python.org/3/library/logging.html#logging.captureWarnings) to forward warnings to the default logging facility.

Reply:
huh, TIL. I will change the log to a `warnings.warn`, give me a minute

Reply:
Changed to a `UserWarning` in [cca4088](https://github.com/appliedAI-Initiative/lakefs-spec/pull/133/commits/cca4088efdc345b42b9bc9f1cc160a297e680334).

Reply:
> huh, TIL.

So did I 😄 But I vaguely remembered there being a separate warning mechanism in the stdlib, which got me on the path towards `warnings.warn`.

Reply:
> Changed to a `UserWarning` in [cca4088](https://github.com/appliedAI-Initiative/lakefs-spec/pull/133/commits/cca4088efdc345b42b9bc9f1cc160a297e680334).

Awesome, this is much cleaner!